metals/src/main/scala/scala/meta/internal/metals/NewFilesProvider.scala

```diff
@@ -66,31 +135,80 @@ class NewFilesProvider(
       )
   }
 
+  private def createWorksheet(directory: Option[AbsolutePath], name: String)(
+      implicit ec: ExecutionContext
+  ): Future[AbsolutePath] = {
+    val path = directory.getOrElse(workspace).resolve(name + ".worksheet.sc")
+    createFile(path)
+  }
+
   private def createFile(
       path: AbsolutePath
-  )(implicit ec: ExecutionContext): Future[URI] = {
+  )(implicit ec: ExecutionContext): Future[AbsolutePath] = {
     val result = Future {
-      Files.createFile(path.toNIO).toUri()
+      AbsolutePath(
+        Files.createFile(path.toNIO)
+      )
     }
     result.onFailure {
-      case NonFatal(e) => scribe.error("Cannot create file", e)
+      case NonFatal(e) =>
+        scribe.error("Cannot create file", e)
+        client.showMessage(
+          MessageType.Error,
+          s"Cannot create file:\n ${e.toString()}"
+        )
     }
     result
   }
 
   private def createFileAndWriteText(path: AbsolutePath, text: String)(
       implicit ec: ExecutionContext
-  ): Future[URI] = {
-    createFile(path).map { newFileUri =>
+  ): Future[AbsolutePath] = {
+    createFile(path).map { _ =>
       path.writeText(text)
-      newFileUri
+      path
     }
   }
 
+  private def openFile(path: AbsolutePath): Unit = {
+    client.metalsExecuteClientCommand(
+      new ExecuteCommandParams(
+        ClientCommands.GotoLocation.id,
+        List(
+          new Location(path.toString(), new org.eclipse.lsp4j.Range()): Object
+        ).asJava
+      )
+    )
+  }
+
   private def classTemplate(kind: String, name: String): String =
     s"""|$kind $name {
         |  
         |}
         |""".stripMargin
 
 }
+
+object DialogEnrichments {
+
+  implicit class DialogContinuation[A](state: Future[Option[A]]) {
+
+    def continueWith[B](
```

Comment:
Looks good, this is almost a MonadTransformer haha.

Maybe `flatMapOption` and `mapOption` ? This will be a bit more natural.

And these methods might be actually useful to add to `MetalsEnrichments`, I remember situations when we were combining Futures and Options.

Reply:
Yeah, I also thought about `OptionT` :D . Done